.github/workflows/docker.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and push Docker image
+
+on:
+  push:
+    branches: [ master, develop ]
+    paths:
+      - 'docker/**'
+  pull_request:
+    branches: [ master, develop ]
+    paths:
+      - 'docker/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Log in to Docker Hub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          username: infinitime
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: ./docker/
+          file: ./docker/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          builder: ${{ steps.buildx.outputs.name }}
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Pushing commits will trigger this workflow and this will be true. I feel like pushing the docker should be a separate workflow that only triggers when adding commits to develop

Reply:
I think I was mistaken about the event_name, however I think this should instead explicitly require a push event rather than anything other than a pull_request, which relies on this workflow to run at intended times only.